### PR TITLE
Adds support for using Secrets

### PR DESCRIPTION
**Pull request by bartlomiejolma**
Adds support for using Secrets

---
Keep in mind: 
- [x] Documentation updates
- [x] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: kedro_airflow_k8s/airflow_dag_template.j2 [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import logging
 
 from airflow import DAG
+from airflow.contrib.kubernetes.secret import Secret
```
Comment (michalzelechowski-getindata) [inline kedro_airflow_k8s/airflow_dag_template.j2:4]: Is it working with Airflow 2.x? I feel like this import is from 1.x only.
EDIT: `from airflow.kubernetes.secret import Secret` is valid for 2.x

File: setup.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "pytest-cov>=2.8.0, <3.0.0",
         "tox==3.21.1",
         "pre-commit==2.9.3",
-        "apache-airflow==2.0.1",
+        "apache-airflow[kubernetes]==2.0.1",
```
Comment (michalzelechowski-getindata) [inline setup.py:23]: Is `kubernetes` required here? I don't see new imports used during tests.

**Review by michalzelechowski-getindata — CHANGES_REQUESTED** [review]

File: kedro_airflow_k8s/airflow_dag_template.j2 (hunk 1 above)
Comment (bartlomiejolma) [reply kedro_airflow_k8s/airflow_dag_template.j2]: Done

File: setup.py (hunk 2 above)
Comment (bartlomiejolma) [reply setup.py:23]: It is imported in the `node_pod`
https://github.com/getindata/kedro-airflow-k8s/pull/71/files#diff-b84725b0af420af10bbf6701b8616b8bc88b57ea4708675a864f651ce9e9eaa9R10
Which is used in `tests/operators/test_node_pod_operator.py`

Comment (michalzelechowski-getindata) [reply setup.py:23]: That's true, it's just that we had this test in the past and it was working correctly without `kubernetes` extra for apache-airflow. If it's removed, the tests should still work, shouldn't they?

Comment (bartlomiejolma) [reply setup.py:23]: I am importing it for the `Secret` type annotation. Without this entry in `setup.py` the test was failing.

**Review by michalzelechowski-getindata — APPROVED** [review]

**Review by szczeles — APPROVED** [review]

**Comment by michalzelechowski-getindata** [comment]
@bartlomiejolma Thank you for the contribution!

**Pull request merged**